src/finally.js

```diff
@@ -3,6 +3,8 @@ module.exports = function(Promise, tryConvertToPromise) {
 var util = require("./util");
 var CancellationError = Promise.CancellationError;
 var errorObj = util.errorObj;
+var NEXT_FILTER = {};
```

Comment:
Because otherwise it's just another object that's not special. [This](https://github.com/petkaantonov/bluebird/blob/master/src/promise.js#L549) makes it special, and this NEXT_FILTER object will fail the check,